chiselFrontend/src/main/scala/Chisel/Assert.scala

```diff
@@ -28,32 +29,36 @@ object assert { // scalastyle:ignore object.name
     * defs need to be compiled first and the SBT project is not set up to do
     * that
     */
-  def apply(cond: Bool, message: String): Unit = macro apply_impl_msg
-  def apply(cond: Bool): Unit = macro apply_impl  // macros currently can't take default arguments
+  // Macros currently can't take default arguments, so we need two functions to emulate defaults.
+  def apply(cond: Bool, message: String): Unit = macro AssertTransform.apply_msg
+  def apply(cond: Bool): Unit = macro AssertTransform.apply
 
-  def apply_impl_msg(c: Context)(cond: c.Tree, message: c.Tree): c.Tree = {
+  def do_apply(cond: Bool, message: String)(sourceInfo: SourceInfo): Unit = macro apply_impl_msg
+  def do_apply(cond: Bool)(sourceInfo: SourceInfo): Unit = macro apply_impl
+
+  def apply_impl_msg(c: Context)(cond: c.Tree, message: c.Tree)(sourceInfo: c.Tree): c.Tree = {
```

Comment:
I rather like the consistency of it. Yes, there's two paths, but that can't be avoided because Scala can't seem to settle on how to do macros with default / named arguments.

The whole chain is complicated (and with source locators, everything becomes more complicated!), but I think each step is easily understandable.